### ISSUE TITLE
Failing Hamcrest Matcher does not print expected value

### ISSUE THREAD

**Issue by bastian-schlueter** [lukas-krecan/JsonUnit#820]
Currently using `json-unit-core:3.4.1`, facing incomplete output on test failure when using a Hamcrest Matcher as a custom Matcher.

### Problem
Given the following assertion:
```java
net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson(
        """
    {
      "someText": "abc123"
    }
    """)
    .withMatcher("exampleMatcher", org.hamcrest.Matchers.equalTo("def456"))
    .isEqualTo(
        """
    {
      "someText": "${json-unit.matches:exampleMatcher}"
    }
    """);
```

I expected the resulting message to contain the `expected` value `def456`.
Instead the output looks like this:
```
JSON documents are different:
Matcher "exampleMatcher" does not match value "abc123" in node "someText". was "abc123"
```

Notice the `. was "abc123"` at the end, which just repeats the `actual` value again.
This can be highly confusing as there is no information in the message, what the `expected` value was.

### Cause
This happens due to a missing call between these two lines: https://github.com/lukas-krecan/JsonUnit/blob/16eb5c21d09dd15b58b1123f3455061eb4c843d9/json-unit-core/src/main/java/net/javacrumbs/jsonunit/core/internal/HamcrestHandler.java#L36-L37

Didn't look too deep into the `Matcher` interface, but looking at existing Hamcrest Matchers it seems like `matcher.describeMismatch(value, description)` only populates the description with the `actual` value.

### Possible solution
Calling `matcher.describeTo(description)` between lines 36 and 37, mentioned above, should allow the Hamcrest Matcher to populate the description with the `expected` value, although I must admit, the formatting looks a little funky due to a missing space character:
```
JSON documents are different:
Matcher "exampleMatcher" does not match value "abc123" in node "someText". "def456"was "abc123"
```

**Comment by lukas-krecan** [comment]
Hi, thanks for report. You are right, [this](https://github.com/lukas-krecan/JsonUnit/pull/839) should fix it

**Comment by lukas-krecan** [comment]
Released as 3.5.0
